processor/marketplace_processor/offer/offer_creation.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+
+
+def handle_offer_creation(create_offer, header, state):
+    """Handle Offer creation.
+
+    Args:
+        create_offer (CreateOffer): The transaction.
+        header (TransactionHeader): The header of the Transaction.
+        state (MarketplaceState): The wrapper around the context.
+
+    Raises:
+        InvalidTransaction
+            - There is already an Offer with the same identifier.
+            - The txn signer does not have an Account.
+            - The source is not a holding.
+            - THe target is not a holding.
+
+    """
+
+    if state.get_offer(identifier=create_offer.id):
+        raise InvalidTransaction(
+            "Failed to create Offer, id {} already exists.".format(
+                create_offer.id))
+
+    if not state.get_account(public_key=header.signer_public_key):
+        raise InvalidTransaction(
+            "Failed to create offer, transaction signer {} does "
+            "not have an Account.".format(header.signer_public_key))
+
+    if create_offer.source and \
+            not state.get_holding(identifier=create_offer.source):
+        raise InvalidTransaction(
+            "Failed to create Offer, Holding id {} listed as source "
+            "does not refer to a Holding.".format(create_offer.source))
+
+    if create_offer.target and \
```

Comment:
As per our discussion on Slack, the TP should also check:
- that the source_quantity is set
- that the target and target_quantity are both set or both unset